xbmc/platform/android/network/NetworkAndroid.cpp

```diff
@@ -349,3 +364,25 @@ void CNetworkAndroid::RetrieveInterfaces()
                 n.toString());
   }
 }
+
+void CNetworkAndroid::onAvailable(const CJNINetwork n)
+{
+  CLog::Log(LOGDEBUG, "CNetworkAndroid::onAvailable The default network is now: {}", n.toString());
+
+  CJNIConnectivityManager connman{CXBMCApp::getSystemService(CJNIContext::CONNECTIVITY_SERVICE)};
+  CJNILinkProperties lp = connman.getLinkProperties(n);
+
+  if (lp)
+  {
+    CJNINetworkInterface intf = CJNINetworkInterface::getByName(lp.getInterfaceName());
+    if (intf)
+      m_defaultInterface = new CNetworkInterfaceAndroid(n, lp, intf);
```

Comment:
I don't see a delete to this new. Maybe use std::unique_ptr instead?

Reply:
I've tried, but this class is common to all platforms and copy operations are done in other parts of the code... with a `delete` would be fine

Reply:
Since m_defaultInterface is local to this class I don't see a problem. You can use .get() to return the raw pointer in functions that need CNetworkInteface*

Reply:
I tried again as you indicated and it works, please tell me if I forgot anything, thanks